Rakefile

```diff
@@ -0,0 +1,128 @@
+  Timeout::Error
+]
+
+desc 'Create a new post file'
+namespace :posts do
+  task :new do
+    # Fetch user command line arguments.
+    pr, host, date = get_cli_options
+
+    # Set default values for host and date if not supplied by user.
+    host ||= 'jnewbery'
+    date ||= next_wednesday
+
+    # Ensure date is valid.
+    unless valid_iso8601_date?(date)
+      puts "Error: Invalid date (#{date} received, YYYY-MM-DD needed). Nothing done, exiting."
+      exit
+    end
+
+    # Fetch pull request data from the GitHub v3 REST API.
+    http = Curl.get("#{GITHUB_API_URL}/#{pr}") { |request| request.headers = headers }
+    response = parse_response(http)
+
+    # Create a new post file if none exists.
+    filename = "#{'_posts/' if File.directory?('_posts')}#{date}-##{pr}.md"
+    if File.file?(filename)
+      puts "Error: Filename #{filename} already exists! Nothing done, exiting."
+      exit
+    else
+      create_post_file!(filename, response, date, host)
+      puts "New file #{filename} created successfully."
+      exit
+    end
+  end
+end
+
+def get_cli_options(options = {})
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: rake posts:new -- <options>'
+    opts.on('--pr NUMBER', 'Pull request number') { |pr| options[:pr] = pr }
+    opts.on('--host USERNAME', "Host's GitHub username") { |host| options[:host] = host }
+    opts.on('--date YYYY-MM-DD', 'Meeting date in ISO8601 format (optional, defaults to next Wednesday)') { |date| options[:date] = date }
+    opts.on('-h', '--help', 'Display this help') do
+      puts opts
+      exit
+    end
+    args = opts.order!(ARGV) {}
+    opts.parse!(args)
+  end
+  [options[:pr], options[:host], options[:date]]
+end
+
+def valid_iso8601_date?(date_string)
+  yyyy, mm, dd = date_string.split('-').map(&:to_i)
+
+  date_string.size == date_string.gsub(/[^0-9-]/i, '').size &&
+    [yyyy, mm, dd].none?(nil) && [yyyy, mm, dd].none?(0) &&
+    yyyy >= Date.today.year && Date.valid_date?(yyyy, mm, dd)
+end
+
+def next_wednesday(date = Date.today, wednesday = 3)
+  date + ((wednesday - date.wday) % 7)
+end
+
+def headers
+  {
+    'accept': 'application/vnd.github.symmetra-preview+json',
+    'user-agent': 'bitcoin-core-review-club.github.io'
+  }
+end
+
+def parse_response(http)
+  if http.response_code == HTTP_SUCCESS
+    JSON.parse(http.body)
+  else
+    puts "HTTP #{http.status}"
+    puts http.body
+    exit
+  end
+rescue *HTTP_ERRORS => e
+  "Error #{e.inspect}"
+end
+
+def create_post_file!(filename, response, date, host)
+  File.open(filename, 'w') do |line|
+    line.puts "---"
+    line.puts "layout: pr"
+    line.puts "date: #{date}"
+    line.puts "title: \"#{response['title']}\""
+    line.puts "pr: #{response['number']}"
+    line.puts "authors: [#{response.dig('user', 'login')}]"
+    line.puts "components: #{parse_components(response['labels'])}"
+    line.puts "created_at: \"#{response["created_at"]}\""
+    line.puts "state: #{response['state']}"
```

Comment:
I'd prefer not to add this (and the `merged` field below), since those properties change. The review club page would become outdated if the state/merged field changed after this script was run the first time.

Reply:
Done.